### DISCUSSION TITLE
Drag and drop files API?

### DISCUSSION THREAD

**Discussion by shenghan97** [neutralinojs/neutralinojs#716] [Ideas]
Hey!

Is it possible in any way to use HTML drag and drop api to access files on the system when the user drop their files into certain area of the app? It seems like browsers’s native api which returns a DataTransfer object doesn’t include the path to the file on local machine. Is there any workaround? If not, I’d love to see that feature in the future! 

Thanks

**Comment by vicnz** [comment]
Is think there is A Browser API for that you can integrate that along with neutralinojs API

**Reply by shenghan97** [reply]
The problem is the browser api (html drag and drop api) doesn’t report the full path of the dropped files, so there’s no way for  neutralinojs to access the files. Unless there’s another way to do things!

**Reply by vicnz** [reply]
you can use something like convert it into a \***data:**\* string ex. `data:SFDFoy96SDIHCEIFUHDFsfcs=` and use the N.js IPC API to write it into a *new* temporary file, then you could use that as the reference, -then remove the temporary file after your operations or when you close the app

I never tried this,  its just a hypothesis

**Reply by shenghan97** [reply]
Does that mean you can only read the file but not write it back? I suppose you should be able to do the same thing with only html apis without native apis. You can always get the content the Data Transfer object but it’s read only. Then you can  make a blob to make temporary changes but can’t save it back to its original location). I probably was not clear before about that my question is I need to read and write access to the local file!

**Reply by robtoll** [reply]
I need the same functionality - drag and drop with acces to all of the file info, including the path.
NWJS does this, but I'd prefer to use Neutralino for the smaller filesize.
Any plans for this in future?

**Reply by davidsky** [reply]
+1 - access to file's full path is an invaluable feature!

It can be used for watching and saving user changes, or partial/preview reads for large files, and much more.

**Reply by mogita** [reply]
+1 on the drag and drop API.

I was trying to implement drag and drop with `react-dropzone` only to find out that Neutralino doesn't set the `path` field like Electron does, so even though `react-dropzone` can already [work with Electron](https://github.com/react-dropzone/file-selector/pull/11/files) (and in theory any underlying framekworks that sets `path` to the full path of the selected file(s)), I still couldn't get the full path which was essential in my use case. I'll need to pass the full path to an external binary as its input.

BTW here's what I could get from the web based `<input />` element, i.e. `react-dropzone`'s base element:

<img width="330" alt="SCR-20230830-szkt" src="https://github.com/neutralinojs/neutralinojs/assets/1173069/f496c86b-b55a-4cb7-81c4-06aa3e53913d">
